dali/pipeline/operators/color/hsv.h

```diff
@@ -0,0 +1,284 @@
+inline void assign_argument_value<CPUBackend>(const OpSpec &spec, const std::string &arg_name,
+                                              argument_t<CPUBackend> &arg) {
+  std::vector<float> tmp;
+  GetSingleOrRepeatedArg(spec, tmp, arg_name);
+  arg = tmp[0];
+}
+
+
+template <>
+inline void assign_argument_value<GPUBackend>(const OpSpec &spec, const std::string &arg_name,
+                                              argument_t<GPUBackend> &arg) {
+  GetSingleOrRepeatedArg(spec, arg, arg_name);
+}
+
+
+template <typename Backend>
+struct WorkspaceInputType {
+  using type = TensorVector<CPUBackend>;
+};
+
+template <>
+struct WorkspaceInputType<GPUBackend> {
+  using type = TensorList<GPUBackend>;
+};
+
+template <typename Backend>
+using workspace_input_t = typename WorkspaceInputType<Backend>::type;
+
+
+}  // namespace hsv
+
+
+
+template <typename Backend>
+class Hsv : public Operator<Backend> {
+ public:
+  explicit Hsv(const OpSpec &spec) :
+          Operator<Backend>(spec),
+          output_type_(spec.GetArgument<DALIDataType>(hsv::kOutputType)) {
+    hsv::assign_argument_value<Backend>(spec, hsv::kHue, hue_);
+    hsv::assign_argument_value<Backend>(spec, hsv::kSaturation, saturation_);
+    hsv::assign_argument_value<Backend>(spec, hsv::kValue, value_);
+    if (std::is_same<Backend, GPUBackend>::value) {
+      kernel_manager_.Resize(1, 1);
+    } else {
+      kernel_manager_.Resize(num_threads_, batch_size_);
+    }
+  }
+
+  
+  ~Hsv() override = default;
+
+
+  DISABLE_COPY_MOVE_ASSIGN(Hsv);
+
+
+ protected:
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+
+  USE_OPERATOR_MEMBERS();
+  hsv::argument_t<Backend> hue_, saturation_, value_;
+  DALIDataType output_type_;
+  kernels::KernelManager kernel_manager_;
+};
+
+
+
+class HsvCpu : public Hsv<CPUBackend> {
+ public:
+  explicit HsvCpu(const OpSpec &spec) : Hsv(spec) {}
+
+  /*
+   * So that compiler wouldn't complain, that
+   * "overloaded virtual function `dali::Operator<dali::CPUBackend>::RunImpl`
+   * is only partially overridden in class `dali::HsvCpu`"
+   */
+  using Operator<CPUBackend>::RunImpl;
+
+  ~HsvCpu() override = default;
+
+
+  DISABLE_COPY_MOVE_ASSIGN(HsvCpu);
+
+
+ protected:
+  bool SetupImpl(std::vector<::dali::OutputDesc> &output_desc,
+                 const workspace_t<CPUBackend> &ws) override;
+
+
+  void RunImpl(Workspace<CPUBackend> &ws) override;
+
+
+ private:
+  mat3 transformation_matrix(float hue, float saturation, float value) const {
+    using namespace hsv;  // NOLINT
+    return Rgb2Yiq * compose_hue(hue) * compose_saturation(saturation) * compose_value(value) *
+           Yiq2Rgb;
+  }
+
+
+  template <typename Kernel, typename InputType>
+  kernels::TensorListShape<> CallSetup(const TensorVector<CPUBackend> &input, int instance_idx) {
+    kernels::KernelContext ctx;
+    kernels::TensorListShape<> sh = input.shape();
+    std::vector<kernels::TensorShape<>> shapes;
+    shapes.resize(sh.num_samples());
```

Comment:
The whole point of TensorList shape is to not have a vector of vectors.
I guess, the dimensionality of the output won't change here, so you can 
```suggestion
    shapes.resize(sh.num_samples(), 3);
```
And than write directly into span the obtained shape.

Reply:
Done